pkg/controllers/management/aks/aks_cluster_handler.go

```diff
@@ -0,0 +1,720 @@
+			return "", to.BoolPtr(true), nil
+		}
+	} else {
+		publicAccess = to.BoolPtr(false)
+	}
+
+	return serviceToken, publicAccess, err
+}
+
+func generateSAToken(restConfig *rest.Config) (string, error) {
+	clientSet, err := kubernetes.NewForConfig(restConfig)
+	if err != nil {
+		return "", fmt.Errorf("error creating clientset: %v", err)
+	}
+
+	return util.GenerateServiceAccountToken(clientSet)
+}
+
+func (e *aksOperatorController) getKubeConfig(cluster *mgmtv3.Cluster) (*rest.Config, error) {
+	ctx := context.Background()
+	restConfig, err := controller.GetClusterKubeConfig(ctx, e.secretsCache, *cluster.Spec.AKSConfig)
+	if err != nil {
+		return nil, err
+	}
+	return restConfig, nil
+}
+
+func (e *aksOperatorController) setUnknown(cluster *mgmtv3.Cluster, condition condition.Cond, message string) (*mgmtv3.Cluster, error) {
+	if condition.IsUnknown(cluster) && condition.GetMessage(cluster) == message {
+		return cluster, nil
+	}
+	cluster = cluster.DeepCopy()
+	condition.Unknown(cluster)
+	condition.Message(cluster, message)
+	var err error
+	cluster, err = e.clusterClient.Update(cluster)
+	if err != nil {
+		return cluster, fmt.Errorf("failed setting cluster [%s] condition %s unknown with message: %s", cluster.Name, condition, message)
+	}
+	return cluster, nil
+}
+
+func (e *aksOperatorController) setTrue(cluster *mgmtv3.Cluster, condition condition.Cond, message string) (*mgmtv3.Cluster, error) {
+	if condition.IsTrue(cluster) && condition.GetMessage(cluster) == message {
+		return cluster, nil
+	}
+	cluster = cluster.DeepCopy()
+	condition.True(cluster)
+	condition.Message(cluster, message)
+	var err error
+	cluster, err = e.clusterClient.Update(cluster)
+	if err != nil {
+		return cluster, fmt.Errorf("failed setting cluster [%s] condition %s true with message: %s", cluster.Name, condition, message)
+	}
+	return cluster, nil
+}
+
+func (e *aksOperatorController) setFalse(cluster *mgmtv3.Cluster, condition condition.Cond, message string) (*mgmtv3.Cluster, error) {
+	if condition.IsFalse(cluster) && condition.GetMessage(cluster) == message {
+		return cluster, nil
+	}
+	cluster = cluster.DeepCopy()
+	condition.False(cluster)
+	condition.Message(cluster, message)
+	var err error
+	cluster, err = e.clusterClient.Update(cluster)
+	if err != nil {
+		return cluster, fmt.Errorf("failed setting cluster [%s] condition %s false with message: %s", cluster.Name, condition, message)
+	}
+	return cluster, nil
+}
+
+// generateValuesYaml generates a YAML string containing any
+// necessary values to override defaults in values.yaml. If
+// no defaults need to be overwritten, an empty string will
+// be returned.
+func generateValuesYaml() (string, error) {
+	values := aksOperatorValues{
+		HTTPProxy:  os.Getenv("HTTP_PROXY"),
+		HTTPSProxy: os.Getenv("HTTPS_PROXY"),
+		NoProxy:    os.Getenv("NO_PROXY"),
+	}
+
+	valuesYaml, err := yaml.Marshal(values)
+	if err != nil {
+		return "", err
+	}
+
+	return string(valuesYaml), nil
+}
```

Comment:
These are also candidates to be moved to a separate package and used by all KEv2 controllers.

Reply:
Ok, it will be reworked